### PR TITLE
fix(FEC-10722): ks missing for external captions

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Issue: ks missing for external captions.
Solution: add ks to external captions.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/k-provider/ovp/external-captions-builder.js [hunk 1]
```diff
@@ -13,14 +13,15 @@ const CaptionsFormatsMap: {[format: string]: string} = {
...
-  static createConfig(captions: Array<Object>): Array<PKExternalCaptionObject> {
+  static createConfig(captions: Array<Object>, ks: string): Array<PKExternalCaptionObject> {
     return captions.map(caption => {
       let url = caption.url;
       let type = CaptionsFormatsMap[caption.format];
       if ([KalturaCaptionType.DFXP, KalturaCaptionType.CAP].includes(caption.format)) {
         url = caption.webVttUrl;
         type = CaptionsFormatsMap[KalturaCaptionType.WEBVTT];
       }
+      url = ks ? url + '/ks/' + ks : url;
```
Comment (yairans) [inline src/k-provider/ovp/external-captions-builder.js:24]: please add test

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/k-provider/ovp/external-captions-builder.js [hunk 2]
```diff
@@ -13,14 +13,15 @@ const CaptionsFormatsMap: {[format: string]: string} = {
 };
 
 class ExternalCaptionsBuilder {
-  static createConfig(captions: Array<Object>): Array<PKExternalCaptionObject> {
+  static createConfig(captions: Array<Object>, ks: string): Array<PKExternalCaptionObject> {
```
Comment (dan-ziv) [inline src/k-provider/ovp/external-captions-builder.js:16]: Add an optional annotation

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/k-provider/ovp/external-captions-builder.js (hunk 1 above)
Comment (OrenMe) [reply src/k-provider/ovp/external-captions-builder.js:24]: need to verify if url already has query string params, e.g. contains `?` and if so add it as `&ks=KS`

File: src/k-provider/ovp/external-captions-builder.js (hunk 2 above)
Comment (Yuvalke) [reply src/k-provider/ovp/external-captions-builder.js:16]: it's not optional, it'll pass empty if it's not exist

File: src/k-provider/ovp/external-captions-builder.js (hunk 1 above)
Comment (Yuvalke) [reply src/k-provider/ovp/external-captions-builder.js:24]: @yairans tests added

Comment (Yuvalke) [reply src/k-provider/ovp/external-captions-builder.js:24]: @OrenMe it worked also as ?ks=ks and /ks/ks

**Review by OrenMe — APPROVED** [review]

**Pull request merged**